### PR TITLE
DBZ-2412 Avoiding NPE in logical router

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.redhat.com/projects/DBZ/issues/DBZ-2412

@rk3rn3r, can you review + merge this one? Best reviewed commit by commit. Thx!

### CODE REVIEW

**Comment by gunnarmorling** [comment]
This should also be back-ported to 1.2.

File: debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java [hunk 1]
```diff
@@ -178,6 +180,8 @@ public void configure(Map<String, ?> props) {
         }
         keyFieldName = config.getString(KEY_FIELD_NAME);
         keyEnforceUniqueness = config.getBoolean(KEY_ENFORCE_UNIQUENESS);
+
+        smtManager = new SmtManager<>(config);
```
Comment (rk3rn3r) [inline debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java:184]: do we need a setter for tests?

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java:184]: I don't think so. The config can be passed in, and I don't see a need for using an alternative impl of that utility class.

**Comment by gunnarmorling** [comment]
Unrelated test failure is tracked at https://issues.redhat.com/browse/DBZ-2398

**Comment by Naros** [comment]
Merged to master & back-ported to 1.2. Thanks!

**Pull request merged**